### ISSUE TITLE
missing understanding automatically load of content

### ISSUE THREAD

**Issue by jochenthomas** [aFarkas/lazysizes#112]
Hi there, I do not understand how lazysizes is loading content (??automatically??) which is close to a page or section. 

In detail:
Given you want to create a scrolling page which has as well a horizontal navigation: How lazysizes detects the section which is close to the current in-view section.
E.g. on the second section of "http://alvarotrigo.com/fullPage" you can slide to the right or left.
It would mean, that the content (css/js/html/svg/...) of "section xx" has to be loaded in advance.
A rule could be: the main sections would be loaded initially (without lazysizes), but the side-content (vertical scroll) will be loaded if the user is one slide ahead or below the relevant vertical content:

main scroll-slide/section 1
main scroll-slide 2
main scroll-slide 3 --> if you are here the vertical content of slide 4.xx has to be loaded (= section 4.1, 4.2, 4.3., ...)
main scroll-slide 4  - vertical section 4.1., vertical section 4.2., vertical section 4.xx
main scroll-slide 5 

Please can you explain how this could work
a) if a js call can be fired if you are e.g. on main scroll-slide 3
b) automatically (as I understand your description = self-initializing, without any configuration or JS callbacks)?

Hope explanation is understandable, thanks in advance for your feedback :--)

**Comment by aFarkas** [comment]
lazysizes is an image lazyloader in the first place, but it is also highly extendable and does detect inview elements automatically.

If you want to load HTML content (and JS, CSS) you can use the [include plugin](https://github.com/aFarkas/lazysizes/tree/gh-pages/plugins/include).

Be aware, in case the include plugin is used you might want to set this option back to default (or not):

```js
window.lazySizesConfig = window.lazySizesConfig || {};
window.lazySizesConfig.preloadAfterLoad = false;
```

Then you can load content like this:

```html
<div class="dynamic-content lazyload" data-include="content.html">
</div>
```

Automatic mean automatic. The content is loaded based on the ``expand``/``expFactor`` option. (How near the content is to the viewport).

Here is a demo, it's easy:
http://embed.plnkr.co/aylEqyvub3PVzmefhk8j/preview

b) If you want to force a preload even if it is not in the viewport (This can become handy in some situation, where a carousel is using cloned elements). You can simply add the class ``lazypreload``:

```js
addEventListener('myevent', function(e){
   $(e.target).next().find('.lazyload').addClass('lazypreload');
});

**Comment by jochenthomas** [comment]
Hi,
that fast fast (too fast for me ;-) ), but new day - new challenges...
O.k. many thanks. Yes the include-module I saw already. The important information for me is: Only IN-VIEW content will be automatically loaded (not pages close to the current position).
So the second option is important (script-based preload)...

Therefore some additional questions :
On page 1 a triggered preload of page 1.1 + 2 + 2.1 + 2.2. and 3 + 3.1  will be initiated. 
Will such preloads queued and loaded step by step according the defined order?
Is there an example in the documentation?

And what happens if the user slides immediately from slide 1 to slide 3 (even if the preload is just at page 2)? 
Will such queue now be stopped and immediately page 3 and 3.1. be started to load?
What happens with the not yet preloaded pages 2.1. + 2.2., will they be (automatically) loaded afterwards?

And a question regarding Google/SEO and pagespeed.
How is the load (I guess the pages are iframes) of lazypreload recognized by Google? At which point the page is completely loaded (given the fact that I will start on page 1 a preload)? As you know this is e.g. important for ranking.

Many thanks again for the fast answer...

**Comment by aFarkas** [comment]
The ``data-include="page2.html|,|page2-1.hmtl css:2-1.css|,|page2.2.html|,|page3.html |,|page3.1.html|"`` is invalid. While you can load multiple JS/CSS files with one ``data-include`` attribute, you can only load one html content per include.

A load will never be aborted. lazySizes also loads 2-3 includes in parallel. The queue works as last in first out. So if there are more than 3 includes initialized/preloaded. The script starts to queue. If the user goes fast to another slide, this slide will be first in queue.

Google bot will load all includes upfront. (no lazyloading) This means your content is fully visible to google.

Please simply try the demo http://embed.plnkr.co/aylEqyvub3PVzmefhk8j/preview and alter its code to your needs.

**Comment by aFarkas** [comment]
I hope this helped, otherwise please re-open.

**Comment by jochenthomas** [comment]
Hi,again thanks for the answer.
In general (without pre-loading scripts) it works, but need to check in more detail (this needs time).Would get in contact later if help is needed.Thanks.
Jochen


 


     Alexander Farkas <notifications@github.com> schrieb am 11:19 Dienstag, 9.Juni 2015:
   

 

Closed #112.—
